truffle/test/LoadContract.js

```diff
@@ -164,46 +163,60 @@ contract('LoadContract', async (accounts) => {
         assert.equal(data.shipment.carrier, CARRIER);
         assert.equal(data.shipment.moderator, MODERATOR);
         assert.equal(data.shipment.state, ShipmentState.CREATED);
-        assert.equal(data.shipment.vaultUri, "");
-        assert.equal(data.shipment.vaultHash, "");
     });
 
     it("should have a getShipmentData function, and work with createNewShipment2", async () => {
         const shipmentUuid = uuidToHex(uuidv4(), true);
-        await contract.createNewShipment2(shipmentUuid, EscrowFundingType.NO_FUNDING, 0, "uri", "hash", CARRIER, {from: SHIPPER});
+        await contract.createNewShipment2(shipmentUuid, EscrowFundingType.NO_FUNDING, 0,  CARRIER, {from: SHIPPER});
+        await notary.registerVault(shipmentUuid, "uri", "hash", {from: SHIPPER});
+
         await contract.setModerator(shipmentUuid, MODERATOR, {from: SHIPPER});
         let data = await getShipmentEscrowData(shipmentUuid);
         assert.equal(data.shipment.shipper, SHIPPER);
         assert.equal(data.shipment.carrier, CARRIER);
         assert.equal(data.shipment.moderator, MODERATOR);
         assert.equal(data.shipment.state, ShipmentState.CREATED);
-        assert.equal(data.shipment.vaultUri, "uri");
-        assert.equal(data.shipment.vaultHash, "hash");
-    })
+    });
 
-    it("should emit VaultHash and set the vaultHash value in the shipment data", async () => {
-        const vaultHash_shipper = "0x1234567890";
-        const vaultHash_carrier = "0x1234567890";
-        const shipmentUuid = await createShipment();
+    it("should allow SHIPPER to update the uri and hash if using createShipment2", async () => {
+        const shipmentUuid = await createShipment2();
 
-        await truffleAssert.reverts(contract.setVaultHash(shipmentUuid, vaultHash_shipper), "Only Shipper or Carrier allowed to set VaultHash");
+        await notary.setVaultUri(shipmentUuid, "new_uri", {from: SHIPPER});
+        await notary.setVaultHash(shipmentUuid, "new_hash", {from: SHIPPER});
 
-        const setVaultTx_shipper = await contract.setVaultHash(shipmentUuid, vaultHash_shipper, {from: SHIPPER});
+        let data = await notary.getVaultNotaryDetails(shipmentUuid);
+        assert.equal(data.vaultUri, "new_uri");
+        assert.equal(data.vaultHash, "new_hash");
+    });
 
-        await truffleAssert.eventEmitted(setVaultTx_shipper, "VaultHash", ev => {
-            return ev.msgSender === SHIPPER  && ev.shipmentUuid === uuidToHex32(shipmentUuid) && ev.vaultHash === vaultHash_shipper;
-        });
-        let data = await getShipmentEscrowData(shipmentUuid);
-        assert.equal(data.shipment.vaultHash, vaultHash_shipper);
+    it("should allow SHIPPER to update the uri and hash if using createShipment", async () => {
```

Comment:
What's the need for having test cases to assert SHIPPER can update the vault notary for both createShipment/createShipment2? Now that the only thing that create2 does is add a carrier, I think that these two test cases are redundant.